pkg/config/viper.go

```diff
@@ -0,0 +1,73 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2018 Datadog, Inc.
+
+package config
+
+import (
+	"io"
+	"strings"
+	"time"
+
+	"github.com/spf13/pflag"
+	"github.com/spf13/viper"
+)
+
+// Config represents an object that can load and store configuration parameters
+// coming from different kind of sources:
+// - defaults
+// - files
+// - environment variables
+// - flags
+type Config interface {
+	Set(key string, value interface{})
+	SetDefault(key string, value interface{})
+	IsSet(key string) bool
+
+	Get(key string) interface{}
+	GetString(key string) string
+	GetBool(key string) bool
+	GetInt(key string) int
+	GetInt64(key string) int64
+	GetFloat64(key string) float64
+	GetTime(key string) time.Time
+	GetDuration(key string) time.Duration
+	GetStringSlice(key string) []string
+	GetStringMap(key string) map[string]interface{}
+	GetStringMapString(key string) map[string]string
+	GetStringMapStringSlice(key string) map[string][]string
+	GetSizeInBytes(key string) uint
+
+	SetEnvPrefix(in string)
+	BindEnv(input ...string) error
+	SetEnvKeyReplacer(r *strings.Replacer)
+
+	UnmarshalKey(key string, rawVal interface{}) error
+	Unmarshal(rawVal interface{}) error
+	UnmarshalExact(rawVal interface{}) error
+
+	ReadInConfig() error
+	ReadConfig(in io.Reader) error
+	MergeConfig(in io.Reader) error
+
+	AllSettings() map[string]interface{}
+
+	AddConfigPath(in string)
+	SetConfigName(in string)
+	SetConfigFile(in string)
+	SetConfigType(in string)
+	ConfigFileUsed() string
+
+	BindPFlag(key string, flag *pflag.Flag) error
+}
+
+// NewConfig returns a new Config object.
+func NewConfig(name string, envPrefix string, envKeyReplacer *strings.Replacer) *viper.Viper {
```

Comment:
Could we mirror the init API from viper or use a struct here ? We might pass more and more parameters to this function.